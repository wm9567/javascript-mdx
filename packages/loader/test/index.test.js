const path = require('path')
const webpack = require('webpack')
const MemoryFs = require('memory-fs')

const testFixture = (fixture, options = {}) => {
  const fileName = `./${fixture}`

  const compiler = webpack({
    context: __dirname,
    entry: `./${fixture}`,
    output: {
      path: path.resolve(__dirname),
      filename: 'bundle.js'
    },
    node: {
      fs: 'empty'
    },
    module: {
      rules: [
        {
          test: /\.md?$/,
          use: [
            {
              loader: 'babel-loader',
              options: {
                presets: ['env', 'stage-0', 'react']
              }
            },
            {
              loader: path.resolve(__dirname, '..')
            }
          ]
        }
      ]
    }
  })

  compiler.outputFileSystem = new MemoryFs()

  return new Promise((resolve, reject) => {
    compiler.run((err, stats) => {
      if (err) reject(err)

      const module = stats.toJson().modules.find(m => m.name === fileName)
        .source

      resolve(module)
    })
  })
}

test(
  'it loads markdown and returns a component',
  async () => {
    await testFixture('fixture.md')
  },
  10000
)
