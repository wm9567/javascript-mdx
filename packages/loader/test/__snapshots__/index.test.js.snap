// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it loads markdown and returns a component 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _tag = require('@mdx-js/tag');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function (_ref) {
  var components = _ref.components;
  return _react2.default.createElement(
    _tag.MDXTag,
    { name: 'wrapper' },
    _react2.default.createElement(
      _tag.MDXTag,
      { name: 'h1', components: components },
      'Hello, world!'
    )
  );
};"
`;
